server/container/guice/cassandra-guice/src/main/java/org/apache/james/modules/mailbox/KeyspacesConfiguration.java

```diff
@@ -0,0 +1,148 @@
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.modules.mailbox;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import org.apache.commons.configuration2.Configuration;
+import org.apache.james.backends.cassandra.init.configuration.KeyspaceConfiguration;
+
+public class KeyspacesConfiguration {
+    public static class Builder {
+        private Optional<String> keyspace;
+        private Optional<String> cacheKeyspace;
+        private Optional<Integer> replicationFactor;
+        private Optional<Boolean> durableWrites;
+
+        public Builder() {
+            keyspace = Optional.empty();
+            cacheKeyspace = Optional.empty();
+            replicationFactor = Optional.empty();
+            durableWrites = Optional.empty();
+        }
+
+        public Builder keyspace(Optional<String> keyspace) {
+            this.keyspace = keyspace;
+            return this;
+        }
+
+        public Builder cacheKeyspace(Optional<String> cacheKeyspace) {
+            this.cacheKeyspace = cacheKeyspace;
+            return this;
+        }
+
+        public Builder cacheKeyspace(String cacheKeyspace) {
+            return cacheKeyspace(Optional.of(cacheKeyspace));
+        }
+
+        public Builder keyspace(String keyspace) {
+            return keyspace(Optional.of(keyspace));
+        }
+
+        public Builder replicationFactor(Optional<Integer> replicationFactor) {
+            this.replicationFactor = replicationFactor;
+            return this;
+        }
+
+        public Builder replicationFactor(int replicationFactor) {
+            return replicationFactor(Optional.of(replicationFactor));
+        }
+
+        public Builder disableDurableWrites() {
+            this.durableWrites = Optional.of(false);
+
+            return this;
+        }
+
+        public KeyspacesConfiguration build() {
+            return new KeyspacesConfiguration(
+                keyspace.orElse(DEFAULT_KEYSPACE),
+                cacheKeyspace.orElse(DEFAULT_CACHE_KEYSPACE),
+                replicationFactor.orElse(DEFAULT_REPLICATION_FACTOR),
+                durableWrites.orElse(true));
+        }
+    }
+
+    public static final String CASSANDRA_KEYSPACE = "cassandra.keyspace";
+    public static final String CASSANDRA_CACHE_KEYSPACE = "cassandra.keyspace.cache";
+    public static final String REPLICATION_FACTOR = "cassandra.replication.factor";
+
+    private static final String DEFAULT_KEYSPACE = "apache_james";
+    private static final String DEFAULT_CACHE_KEYSPACE = "apache_james_cache";
+    private static final int DEFAULT_REPLICATION_FACTOR = 1;
```

Comment:
idem

Reply:
Yes. I don't ame at altering already choosen default values.

Also, we want things to work "by default" on a single node cassandra, which implies a replication factor of 1.